widgetssdk/src/main/java/com/glia/widgets/engagement/EngagementRepositoryImpl.kt

```diff
@@ -531,23 +530,25 @@ internal class EngagementRepositoryImpl(
         currentEngagement?.also {
             currentEngagement = null
             unsubscribeFromEvents(it)
-            notifyEngagementEnded(it)
-            resetState()
+            notifyEngagementEnded(it, true)
+
+            resetState(it.state.actionOnEnd == ActionOnEnd.RETAIN)
 
-            if (it is OmnicoreEngagement) {
-                fetchSurvey(it, true)
+            if (it is OmnicoreEngagement && it.state.actionOnEnd == ActionOnEnd.SHOW_SURVEY) {
+                fetchSurvey(it)
             } else {
                 _survey.onNext(SurveyState.Empty)
             }
         }
     }
 
-    private fun notifyEngagementEnded(engagement: Engagement) {
-        val state = if (engagement is OmnibrowseEngagement)
-            State.FinishedCallVisualizer
+    private fun notifyEngagementEnded(engagement: Engagement, isEndedByOperator: Boolean) {
+        val engagementType = if (engagement is OmnibrowseEngagement)
+            EngagementType.CallVisualizer
         else
-            State.FinishedOmniCore
+            EngagementType.OmniCore
 
+        val state = State.EngagementEnded(engagementType, !isEndedByOperator, engagement.state.actionOnEnd)
```

Comment:
I'd like to suggest changing the `isEndedByVisitor` param in `State.EngagementEnded` to `isEndedByOperator`. By doing this, we will deal with the same logical entity tight to operator. I think it will improve readability, simplify the code and drop the negation in the `!isEndedByOperator` like here we see.

Reply:
Every data model should correctly represent data that matters. And in this class, we only care whether the engagement has been ended by the visitor or not. For example, we don't care if it was ended by the operator or ended because of some error. We only care if it was an explicit action by the visitor to end the engagement because in this case, we don't need to show any alert dialogs.

Overall I agree that it is subjective depending on which way you look at it. And your argument also makes sense to me but I would prefer to leave it this way unless you have another argument )

Reply:
Okay. I will not insist in this case.